term/src/terminalstate/mod.rs

```diff
@@ -1979,7 +1979,11 @@ impl TerminalState {
                 checksum += u16::from(ch as u8);
             }
         }
-        checksum
+        if checksum == 0 {
```

Comment:
Thanks for this!

Please add a comment here; basically the description from the PR plus a reference to this PR, so that it is easier for someone in the future to understand why this is here!